pkgs/by-name/fi/firefly-iii/package.nix

```diff
@@ -1,52 +1,84 @@
 { lib
 , fetchFromGitHub
-, buildNpmPackage
+, stdenvNoCC
+, nodejs
+, fetchNpmDeps
+, buildPackages
 , php83
 , nixosTests
+, nix-update-script
 , dataDir ? "/var/lib/firefly-iii"
 }:
 
 let
   pname = "firefly-iii";
   version = "6.1.18";
   phpPackage = php83;
+  npmDepsHash = "sha256-MoxkNxfVeIFkiNVzfehQ9FpC65kBj8ZmvwaRf4MVRIg=";
 
   src = fetchFromGitHub {
     owner = "firefly-iii";
     repo = "firefly-iii";
     rev = "v${version}";
     hash = "sha256-mA7gvKhHouUUz1Aix7253O/+VcufoEFwdcJeZxnazEo=";
   };
-
-  assets = buildNpmPackage {
-    pname = "${pname}-assets";
-    inherit version src;
-    npmDepsHash = "sha256-MoxkNxfVeIFkiNVzfehQ9FpC65kBj8ZmvwaRf4MVRIg=";
-    dontNpmBuild = true;
-    installPhase = ''
-      runHook preInstall
-      npm run prod --workspace=v1
-      npm run build --workspace=v2
-      cp -r ./public $out/
-      runHook postInstall
-    '';
-  };
 in
 
-phpPackage.buildComposerProject (finalAttrs: {
+stdenvNoCC.mkDerivation (finalAttrs: {
   inherit pname src version;
 
+  buildInputs = [ phpPackage ];
+
+  nativeBuildInputs = [
+    nodejs
+    nodejs.python
+    buildPackages.npmHooks.npmConfigHook
+    phpPackage.composerHooks.composerInstallHook
+    phpPackage.packages.composer-local-repo-plugin
+  ];
+
+  composerNoDev = true;
+  composerNoPlugins = true;
+  composerNoScripts = true;
+  composerStrictValidation = true;
+  strictDeps = true;
+
   vendorHash = "sha256-EpMypgj6lZDz6T94bGoCUH9IVwh7VB4Ds08AcCsreRw=";
 
+  npmDeps = fetchNpmDeps {
+    inherit src;
+    name = "${pname}-npm-deps";
+    hash = npmDepsHash;
+  };
+
+  composerRepository = phpPackage.mkComposerRepository {
+    composer = phpPackage.packages.composer-local-repo-plugin;
```

Comment:
Are you sure about requiring this? Normally it should be automatically provided, see: https://github.com/drupol/nixpkgs/blob/d083aa788eb7c24c121ce2a2f9ca58350eaa7235/pkgs/build-support/php/builders/v1/build-composer-repository.nix#L25-L26

Reply:
Fair enough. This is a remnant from back when I was trying to make this work. Changed!